### PR TITLE
Support for joining and leaving public channels

### PR DESCRIPTION
**Pull request by squallstar**
img

### CODE REVIEW

File: js/build.js [hunk 1]
```diff
@@ -637,6 +669,56 @@ Fliplet.Widget.instance('chat', function (data) {
...
+      fetchChatChannels = null;
+
+      return getContacts(false).then(function() {
+        return getConversations(false);
+      }).then(function () {
+        chat.poll();
+
+        channelsSelected = [];
+        $('.contacts-done-holder').removeClass('creating');
+        $('[name="group-tabs"]:eq(0)').click();
```
Comment (tonytlwu) [inline js/build.js]: Wrap this line to something liek a `switchToDM()` function? Makes it easier if DOM changes.

I wonder if `$('.contacts-done-holder').removeClass('creating');` can be wrapped to a function with a specific purpose too.

File: js/build.js [hunk 2]
```diff
@@ -1904,7 +2054,7 @@ Fliplet.Widget.instance('chat', function (data) {
       }
 
       $('.chat-card[data-conversation-id="' + data.id + '"]').replaceWith(conversationHTML);
-      return 
+      return
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
      return;
```

File: js/build.js (hunk 1 above)
Comment (squallstar) [reply js/build.js]: doen

File: templates/group-tabs.build.hbs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<section class="group-tabs-holder">
+  <input type="radio" name="group-tabs" checked="checked" id="group-tabs-0" value="" />
+  <label for="group-tabs-0">Direct messages</label>
+  <input type="radio" name="group-tabs" id="group-tabs-1" value="1" />
+  <label for="group-tabs-1">Public channels</label>
+</section>
```
Comment (tonytlwu) [inline templates/group-tabs.build.hbs]: https://jsfiddle.net/tonytlwu/rs1ebv8j/ is ready to go. You could swap it in now (though I'm wondering whether the CSS part should be added to a `fliplet-api` package. Maybe in `bootstrap-css`.

**Comment by tonytlwu** [comment]
1. Does the search bar run the search across both tabs?
1. Is there a placeholder message for when there is no search result?

**Comment by squallstar** [comment]
@tonytlwu 
1. yes
2. channels only appear when there's at least one. It's all automatic based on whether the app has public channels available!

**Comment by squallstar** [comment]
@tonytlwu css changes done
screenshot 2019-02-08 at 10 45 44

**Comment by squallstar** [comment]
@tonytlwu when you get to this I've also noticed that there's an unwanted horizontal scrollbar:

image

You can check this on staging for app 24353.

**Comment by tonytlwu** [comment]
@squallstar Thanks. I'll look into it. Looks like there's no way it'll fit in all device widths without further tweaks. I'll deal with it.

Example:

iPhone XS Max

(Also, what's `undefined` doing here? 🤔 )

image

iPhone 5

image

**Comment by squallstar** [comment]
@tonytlwu I'm checking that `undefined` already, I didn't have that on local; probably related to different fields configuration for title, fullName, etc.

**Pull request merged**